### ISSUE TITLE
rnp_op_verify_execute should process a literal packet (tag 11)

### ISSUE THREAD

**Issue by kaie** [rnpgp/rnp#1191]
I need support for a scenario, which is fixed by removing the following check, which prevents the processing of unexpected literal data.

```
diff --git a/third_party/rnp/src/librepgp/stream-parse.cpp b/third_party/rnp/src/librepgp/stream-parse.cpp
--- a/third_party/rnp/src/librepgp/stream-parse.cpp
+++ b/third_party/rnp/src/librepgp/stream-parse.cpp
@@ -2202,12 +2202,14 @@ init_packet_sequence(pgp_processing_ctx_
             ret = init_compressed_src(&psrc, lsrc);
             break;
         case PGP_PKT_LITDATA:
+/*
             if ((lsrc->type != PGP_STREAM_ENCRYPTED) && (lsrc->type != PGP_STREAM_SIGNED) &&
                 (lsrc->type != PGP_STREAM_COMPRESSED)) {
                 RNP_LOG("unexpected literal pkt");
                 ret = RNP_ERROR_BAD_FORMAT;
                 break;
             }
+*/
             ret = init_literal_src(&psrc, lsrc);
             break;
         default:
```

The scenario uses a combination of RNP and GnuPG for decoding a message. The message is encrypted, the secret key is on a smartcard. We detect that RNP and its keys cannot decrypt the message. We try to use GnuPG to decrypt the message.

However, we don't want GnuPG to perform the signature verification, therefore we use GnuPG with a special parameter "--unwrap", which removes the encryption layer, but which keeps other encoding, such as signatures.

(Actually we use the GPGME library with parameter GPGME_DECRYPT_UNWRAP, but that simply forwards to gpg --unwrap.)

After we have removed the encryption layer, GnuPG gives us the remaining OpenPGP data packets, which still need processing. To process them, I call rnp_op_verify_execute. In most scenarios, this works fine.

Today I ran into a scenario, where the results of decryption is simply one "literal data packet (tag 11)". For that input, rnp_op_verify_execute fails with RNP_ERROR_BAD_FORMAT, apparently because of the quoted check above.

Would you be able to allow this to work?

Maybe you could handle this scenario specially? I'm guessing whenever I need the decoding of PGP_PKT_LITDATA to work, it is probably the only packet present.

**Comment by ni4** [comment]
@kaie Thanks for reporting, actually didn't thought that this functionality will ever be needed. Will need some time to think about the best way of implementing this, most likely just removing mentioned checks should work well.

**Comment by ni4** [comment]
Closing as it was fixed in PR #1193
